sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp

```diff
@@ -35,41 +36,22 @@ namespace Azure { namespace Core { namespace Http {
     constexpr static size_t DefaultUploadChunkSize = 1024 * 64;
     constexpr static size_t MaximumUploadChunkSize = 1024 * 1024;
 
-    struct HandleManager final
-    {
-      Context const& m_context;
-      Request& m_request;
-      HINTERNET m_connectionHandle;
-      HINTERNET m_requestHandle;
-
-      HandleManager(Request& request, Context const& context)
-          : m_request(request), m_context(context)
-      {
-        m_connectionHandle = NULL;
-        m_requestHandle = NULL;
-      }
-
-      ~HandleManager()
+    // unique_ptr class wrapping an HINTERNET handle
+    class HINTERNET_deleter {
```

Comment:
Nice cleanup :)

Reply:
I had to tease the HandleManager apart to enable passing it into the WinHTTP WebSocket transport. And eventually it went away in favor of a class derived from std::unique_ptr. I was happy at how little code was actually changed).